scapy/contrib/cansocket_native.py

```diff
@@ -74,6 +79,47 @@ def __init__(self, channel=None, receive_own_messages=False,
         self.ins.bind((self.channel,))
         self.outs = self.ins
 
+    def command(self):
+        d = dict(
+            channel=self.channel,
+            receive_own_messages=self.receive_own_messages,
+            can_filters=self.can_filters,
+            remove_padding=self.remove_padding,
+            basecls=self.basecls
+        )
+
+        dnv = []
+        for dn, dv in six.iteritems(d):
+            if (dv is None or
+                    (isinstance(dv, (list, dict, set)) and len(dv) == 0)):
+                continue
+            elif dn == "basecls":
+                dv = (
+                    dv.__name__
+                    if hasattr(dv, "__name__")
+                    else dv.__class__.__name__
+                )
+            elif dn == "can_filters":
+                def can_filter_g(d):
+                    for fn, fv in six.iteritems(d):
+                        yield "'%s':%s" % (fn, fv)
+
+                dv = "[%s]" % ",".join(
+                    "{%s}" % ",".join(
+                        can_filter_g(filter)
+                    ) for filter in dv
+                )
+            else:
+                dv = repr(dv)
+            dnv.append("%s=%s" % (dn, dv))
+
+        clsname = (
```

Comment:
you should use `self.__class__.__name__` here. I don't see a case where the class name is in `self.__name__`